### PR TITLE
Feature/EnemyScriptableObjects

### PR DESCRIPTION
**Pull request by Shadowmage77**
- Turned enemy-controlling stats into a scriptable objects called AiStats
- Enemy Poolers created for all 5 current enemy types
- TrueScene changes are just lots of references and prefab updates for enemies and enemySpawner

### CODE REVIEW

File: Assets/Code/Scripts/EditorObject/AiStats.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using System.Collections.Generic;
+using Generic;
+using UnityEngine;
+
+namespace EditorObject
+{
+    [CreateAssetMenu(menuName = "Enemy/AI Stats", fileName = "New AI Stats")]
+    public class AiStats : ScriptableObject, IPoolableInstantiateData
+    {
+        // TODO: Enemy type
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/AiStats.cs:11]: Do we want to add this here in this PR or add another ticket?

File: Assets/Code/Scripts/EditorObject/AiStats.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+using System.Collections;
+using System.Collections.Generic;
+using Generic;
+using UnityEngine;
+
+namespace EditorObject
+{
+    [CreateAssetMenu(menuName = "Enemy/AI Stats", fileName = "New AI Stats")]
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/AiStats.cs:8]: Switch to either:
`[CreateAssetMenu(menuName = "AI Stats", fileName = "New AI Stats")]`
or
`[CreateAssetMenu(menuName = "EditorObject/AI Stats", fileName = "New AI Stats")]`

File: Assets/Code/Scripts/EditorObject/AiStats.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        /// <summary>
+        /// Multiplier of the movementGroup top speed
+        /// </summary>
+        [SerializeField] private float gearModifier = 0.8f;
+
+        /// <summary>
+        /// The maximum force determines how quickly an enemy groups and separates
+        /// </summary>
+        [SerializeField] private float maxForce = 1;
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/AiStats.cs:31]: Rename to something like `MaxMovementForce`

File: Assets/Code/Scripts/EditorObject/AiStats.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+        /// <summary>
+        /// The amount of danger level score an enemy gives upon death
+        /// </summary>
+        [SerializeField] private int dlScore = 10;
+
+        /// <summary>
+        /// The distance from the target an enemy has to be to start counting down to attack
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/AiStats.cs:39]: Mention the unit is meters

File: Assets/Code/Scripts/EditorObject/AiStats.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        /// </summary>
+        [SerializeField] private float attackRange = 15;
+
+        /// <summary>
+        /// The distance away from the player that an enemy will follow when Chasing
+        /// </summary>
+        [SerializeField] private float followRange = 10;
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/AiStats.cs:46]: ditto unit

File: Assets/Code/Scripts/Enemies/VehicleAI.cs [hunk 6]
```diff
@@ -110,7 +111,7 @@ public override void Die()
                                 Random.Range(-minMaxTorque, minMaxTorque)),
                                 ForceMode.Impulse);
         vehicleController.enabled = false;
-        Instantiate(itemDrop, this.transform.position, Quaternion.identity);
+        // Instantiate(itemDrop, this.transform.position, Quaternion.identity);
```
Comment (toryjleo) [inline Assets/Code/Scripts/Enemies/VehicleAI.cs:114]: delete line

**Review by toryjleo — CHANGES_REQUESTED** [review]

File: Assets/Code/Scripts/EditorObject/AiStats.cs (hunk 1 above)
Comment (Shadowmage77) [reply Assets/Code/Scripts/EditorObject/AiStats.cs:11]: This is in, just an old comment I missed.

File: Assets/Code/Scripts/Enemies/Ai.cs [hunk 7]
```diff
@@ -8,39 +9,35 @@
...
     internal float timeByTarget = 0;
 
+    // public bool canAim = false;
+    // public float StartingHP;
+    // public float maxForce;
+    // public float dlScore;
+    // protected float attackRange = 15;
+    // public float minimumRange;
+    // internal float TIME_BY_TARGET_TO_ATTACK;
+
```
Comment (toryjleo) [inline Assets/Code/Scripts/Enemies/Ai.cs:31]: Please remove

File: Assets/Code/Scripts/Enemies/Ai.cs [hunk 8]
```diff
@@ -8,39 +9,35 @@
...
     public bool inWorld = false;
 
     //Each enemy's speed is relative to a player's gear
-    [SerializeField] public int movementGroup;
+    // [SerializeField] public int movementGroup;
     //Top speed is determined as a percentage of the gear's max speed
-    [SerializeField] public float gearModifier;
+    // [SerializeField] public float gearModifier;
```
Comment (toryjleo) [inline Assets/Code/Scripts/Enemies/Ai.cs:37]: remove

File: Assets/Code/Scripts/Enemies/Ai.cs [hunk 9]
```diff
@@ -138,7 +140,7 @@ public void Spawn(Vector3 position, Vector3 playerLocation)
         transform.rotation = Quaternion.LookRotation(playerLocation - position);
         gameObject.SetActive(true);
 
-        TIME_BY_TARGET_TO_ATTACK = 2.0f;
+        // TIME_BY_TARGET_TO_ATTACK = 2.0f;
```
Comment (toryjleo) [inline Assets/Code/Scripts/Enemies/Ai.cs:143]: remove

**Review by toryjleo — CHANGES_REQUESTED** [review]

**Review by toryjleo — APPROVED** [review]

**Pull request merged**